### PR TITLE
Fix citizen spawning on wakeup not in enough space.

### PR DESCRIPTION
**Pull request by marchermans**
Fixes citizen spawning on top of the bed if there is not enough space.

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 1]
```diff
@@ -989,12 +992,12 @@ private void cleanupChatMessages()
     @Override
     public void setCustomNameTag(final String name)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:993]: MAJOR Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 2]
```diff
@@ -83,23 +83,26 @@
  */
 public class EntityCitizen extends EntityAgeable implements INpc
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:84]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 66 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "EntityCitizen" has 103 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 3]
```diff
@@ -83,23 +83,26 @@
...
 {
-    private static final float CONST_BED_HEIGHT = 0.6875f;
-    private static final float CONST_HALF_BLOCK = 0.5f;
+    private static final float CONST_BED_HEIGHT             = 0.6875f;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:86]: MINOR Upper-case this literal "f" suffix. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 4]
```diff
@@ -1503,7 +1507,9 @@ public void trySleep(final BlockPos bedLocation)
             return;
         }
 
-        this.setPosition((double)((float)bedLocation.getX() + CONST_HALF_BLOCK), (double)((float)bedLocation.getY() + CONST_BED_HEIGHT), (double)((float)bedLocation.getZ() + CONST_HALF_BLOCK));
+        this.setPosition((double) ((float) bedLocation.getX() + CONST_HALF_BLOCK),
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:1510]: MINOR Use a "double" or "BigDecimal" instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 5]
```diff
@@ -83,23 +83,26 @@
  */
 public class EntityCitizen extends EntityAgeable implements INpc
 {
-    private static final float CONST_BED_HEIGHT = 0.6875f;
-    private static final float CONST_HALF_BLOCK = 0.5f;
+    private static final float CONST_BED_HEIGHT             = 0.6875f;
+    private static final float CONST_HALF_BLOCK             = 0.5f;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:87]: MINOR Upper-case this literal "f" suffix. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 6]
```diff
@@ -1503,7 +1507,9 @@ public void trySleep(final BlockPos bedLocation)
             return;
         }
 
-        this.setPosition((double)((float)bedLocation.getX() + CONST_HALF_BLOCK), (double)((float)bedLocation.getY() + CONST_BED_HEIGHT), (double)((float)bedLocation.getZ() + CONST_HALF_BLOCK));
+        this.setPosition((double) ((float) bedLocation.getX() + CONST_HALF_BLOCK),
+          (double) ((float) bedLocation.getY() + CONST_BED_HEIGHT),
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:1511]: MINOR Use a "double" or "BigDecimal" instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 7]
```diff
@@ -83,23 +83,26 @@
  */
 public class EntityCitizen extends EntityAgeable implements INpc
 {
-    private static final float CONST_BED_HEIGHT = 0.6875f;
-    private static final float CONST_HALF_BLOCK = 0.5f;
+    private static final float CONST_BED_HEIGHT             = 0.6875f;
+    private static final float CONST_HALF_BLOCK             = 0.5f;
     private static final float CONST_SLEEPING_RENDER_OFFSET = -1.5f;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:88]: MINOR Upper-case this literal "f" suffix. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 8]
```diff
@@ -1503,7 +1507,9 @@ public void trySleep(final BlockPos bedLocation)
             return;
         }
 
-        this.setPosition((double)((float)bedLocation.getX() + CONST_HALF_BLOCK), (double)((float)bedLocation.getY() + CONST_BED_HEIGHT), (double)((float)bedLocation.getZ() + CONST_HALF_BLOCK));
+        this.setPosition((double) ((float) bedLocation.getX() + CONST_HALF_BLOCK),
+          (double) ((float) bedLocation.getY() + CONST_BED_HEIGHT),
+          (double) ((float) bedLocation.getZ() + CONST_HALF_BLOCK));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:1512]: MINOR Use a "double" or "BigDecimal" instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 9]
```diff
@@ -1535,7 +1541,7 @@ public float getRenderOffsetX()
             return 0;
         }
 
-        return CONST_SLEEPING_RENDER_OFFSET * (float)enumfacing.getFrontOffsetX();
+        return CONST_SLEEPING_RENDER_OFFSET * (float) enumfacing.getFrontOffsetX();
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:1544]: MINOR Use a "double" or "BigDecimal" instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 10]
```diff
@@ -1554,7 +1560,7 @@ public float getRenderOffsetZ()
             return 0;
         }
 
-        return CONST_SLEEPING_RENDER_OFFSET * (float)enumfacing.getFrontOffsetZ();
+        return CONST_SLEEPING_RENDER_OFFSET * (float) enumfacing.getFrontOffsetZ();
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:1563]: MINOR Use a "double" or "BigDecimal" instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 11]
```diff
@@ -83,23 +83,26 @@
...
-    private static final DataParameter<Integer> DATA_IS_FEMALE       = EntityDataManager.<Integer>createKey(EntityCitizen.class, DataSerializers.VARINT);
-    private static final DataParameter<Integer> DATA_COLONY_ID       = EntityDataManager.<Integer>createKey(EntityCitizen.class, DataSerializers.VARINT);
-    private static final DataParameter<Integer> DATA_CITIZEN_ID      = EntityDataManager.<Integer>createKey(EntityCitizen.class, DataSerializers.VARINT);
-    private static final DataParameter<String>  DATA_MODEL           = EntityDataManager.<String>createKey(EntityCitizen.class, DataSerializers.STRING);
-    private static final DataParameter<String>  DATA_RENDER_METADATA = EntityDataManager.<String>createKey(EntityCitizen.class, DataSerializers.STRING);
-    private static final DataParameter<Boolean> DATA_IS_ASLEEP = EntityDataManager.createKey(EntityCitizen.class, DataSerializers.BOOLEAN);
-    private static final DataParameter<BlockPos> DATA_BED_POS = EntityDataManager.createKey(EntityCitizen.class, DataSerializers.BLOCK_POS);
+    private static final int SPAWN_SEARCH_SIZE = 2;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:94]: MAJOR Remove this unused "SPAWN_SEARCH_SIZE" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 85 issues
* ![MAJOR][MAJOR] 74 major
* ![MINOR][MINOR] 10 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/09b447c533c8f6b892f383b8fab46646eb4b21e9/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by DjTylerD** [comment]
thanks for the quick update, which was planned tomorrow
towhall level 5 resolved
I continue another test and see to propose sugestions

**Pull request merged**